saleor/tests/e2e/orders/test_order_calculate_simple_tax_based_on_shipping_address_tax_class.py

```diff
@@ -0,0 +1,245 @@
+        permission_manage_orders,
+    ]
+    assign_permissions(e2e_staff_api_client, permissions)
+
+    (
+        warehouse_id,
+        channel_id,
+        channel_slug,
+        shipping_method_id,
+        shipping_price,
+    ) = prepare_shop_with_few_shipping_zone_countries(e2e_staff_api_client, "6.66")
+
+    shipping_country_tax_rate = prepare_tax_configuration(
+        e2e_staff_api_client,
+        channel_slug,
+        shipping_country_code="DE",
+        shipping_country_tax_rate=19,
+        billing_country_code="CZ",
+        billing_country_tax_rate=21,
+    )
+
+    variant_price = "155.88"
+    (_product_id, product_variant_id, product_variant_price) = prepare_product(
+        e2e_staff_api_client,
+        warehouse_id,
+        channel_id,
+        variant_price,
+    )
+
+    # Step 1 - Create a draft order
+    input = {
+        "channelId": channel_id,
+    }
+    data = draft_order_create(
+        e2e_staff_api_client,
+        input,
+    )
+    order_id = data["order"]["id"]
+    product_variant_price = float(product_variant_price)
+
+    # Step 2 - Add lines to the order
+    lines = [{"variantId": product_variant_id, "quantity": 1}]
+    order_data = order_lines_create(
+        e2e_staff_api_client,
+        order_id,
+        lines,
+    )
+    order_data = order_data["order"]
+    assert order_data["total"]["gross"]["amount"] == product_variant_price
+
+    # Step 3 - Add shipping and billing addresses
+    shipping_address = {
+        "firstName": "John",
+        "lastName": "Muller",
+        "companyName": "Saleor Commerce DE",
+        "streetAddress1": "Potsdamer Platz 47",
+        "streetAddress2": "",
+        "postalCode": "85131",
+        "country": "DE",
+        "city": "Pollenfeld",
+        "phone": "+498421499469",
+        "countryArea": "",
+    }
+    billing_address = {
+        "firstName": "John",
+        "lastName": "Muller",
+        "companyName": "Saleor Commerce CZ",
+        "streetAddress1": "Sluneční 1396",
+        "streetAddress2": "",
+        "postalCode": "74784",
+        "country": "CZ",
+        "city": "Melc",
+        "phone": "+420722274643",
+        "countryArea": "",
+    }
+    input = {
+        "userEmail": "test_user@test.com",
+        "shippingAddress": shipping_address,
+        "billingAddress": billing_address,
+    }
+    draft_order = draft_order_update(
+        e2e_staff_api_client,
+        order_id,
+        input,
+    )
+    assert draft_order["order"]["userEmail"] == "test_user@test.com"
+    assert draft_order["order"]["shippingAddress"] is not None
+    assert draft_order["order"]["billingAddress"] is not None
+
+    # Step 3 - Add a shipping method
+    input = {"shippingMethod": shipping_method_id}
+    order_data = draft_order_update(
+        e2e_staff_api_client,
+        order_id,
+        input,
+    )
+    order_data = order_data["order"]
+    shipping_price = order_data["shippingPrice"]["gross"]["amount"]
```

Comment:
When you change to prices_entered_with_tax=False then here `net` price should be asserted